File: java/UMLGraph/src/org/daisy/urakawa/properties/channel/ChannelDoesNotExistException.java
package org.daisy.urakawa.properties.channel;

import org.daisy.urakawa.exceptions.CheckedException;

/**
 * This exception should be thrown when trying to remove a Channel
 * which name does not exist in the list of current channel.
 */
public class ChannelDoesNotExistException extends CheckedException {

	/**
	 * 
	 */
	private static final long serialVersionUID = 5910996831315559009L;
}

File: csharp/core/exception/ChannelDoesNotExistException.cs
using System;

namespace urakawa.exception
{
    /// <summary>
    /// This exception should be thrown when trying to remove a Channel
    /// whose localName does not exist in the list of current channels.
    /// </summary>
    public class ChannelDoesNotExistException : CheckedException
    {
        /// <summary>
        /// Constructor setting the message of the exception
        /// </summary>
        /// <param name="msg">The message</param>
        public ChannelDoesNotExistException(string msg) : base(msg)
        {
        }

        /// <summary>
        /// Constructor setting the message and inner <see cref="Exception"/> of the exception
        /// </summary>
        /// <param name="msg">The message</param>
        /// <param name="inner">The inner exception</param>
        public ChannelDoesNotExistException(string msg, Exception inner) : base(msg, inner)
        {
        }
    }
}